### PR TITLE
Update DG links and fix formatting issues

### PR DESCRIPTION
**Pull request by Weiennn**

### CODE REVIEW

**Review by aexolate — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -102,7 +110,7 @@ How the `Logic` component works:
 
 1. When `Logic` is called upon to execute a command, it is passed to an `AddressBookParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
 1. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `DeleteCommand`) which is executed by the `LogicManager`.
-1. The command can communicate with the `Model` when it is executed (e.g. to delete a person).
+1. The command can communicate with the `Model` when it is executed (e.g. to delete a person and to commit the current state of the addressbook for `undo` and `redo` commands).
```
Comment (Goh-Li-Ting) [inline docs/DeveloperGuide.md:113]: Would it be clearer to write "commit the current state of the addressbook to be used for `undo` and `redo` commands"? I am worried that some people may misinterpret that committing the current state of the addressbook only happens for `undo` and `redo` commands.

**Review by ricketytoc — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Weiennn) [reply docs/DeveloperGuide.md]: yep that makes sense, changed!

**Pull request merged**